feat/coding style (#8)
ovh/design-system: packages/specifications/principles/coding-style.mdx

import OdsCodingStyleTesting from '@ovhcloud/ods-specifications/principles/coding-style-testing.mdx';

# Component coding style

## General rules

To achieve consistency in OVHcloud Design System, there are some rules regarding code standards we would like you to follow.

- [Linters / formatters](#linters-and-formatters)
- [Controllers](#controllers)
- [Typescript](#typescript)
    - [any type](#any-type)
    - [TSDoc comments](#tsdoc-comments)
- [ODS component dependencies](#ods-component-dependencies)
- [SCSS](#scss)
- [Testing](#testing)
    - [Types of tests](#types-of-tests)
        - [Unit testing](#unit-testing)
        - [End-to-end testing](#end-to-end-testing)
        - [Visual regression testing](#visual-regression-testing)

### Linters and formatters

As automated process of analyzing code, we are using ESLint with `typescript-eslint` for linting.
For code formatting, we are using Prettier.

### Controllers

> We used to move logic to controllers so this logic can be used by the different implementations.

Reviewer: ```suggestion
Logic has to be done in controllers so it can be used by the different implementations.
```
